Userland/Utilities/uniq.cpp

```diff
@@ -5,75 +5,127 @@
  */
 
 #include <AK/RefPtr.h>
+#include <AK/StringView.h>
 #include <LibCore/ArgsParser.h>
+#include <LibCore/Stream.h>
 #include <LibCore/System.h>
-#include <errno.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
 #include <unistd.h>
 
-struct linebuf {
-    char* buf = NULL;
-    size_t len = 0;
-};
+static Optional<String> get_line_content(StringView line, size_t count, bool show_duplicates, bool print_count)
+{
+    if (!show_duplicates || count > 0) {
+        if (print_count)
+            return String::formatted("{} {}\n", count + 1, line);
+        else
+            return String::formatted("{}\n", line);
+    }
+    return {};
+}
 
-static FILE* get_stream(char const* filepath, char const* perms)
+static StringView skip(StringView buf, unsigned int nchars, unsigned int nfields)
 {
-    FILE* ret;
+    StringView tmp(buf);
+    tmp = tmp.trim("\n"sv);
 
-    if (filepath == nullptr) {
-        if (perms[0] == 'r')
-            return stdin;
-        return stdout;
+    if (nfields) {
+        bool in_field = false;
+        auto fields = tmp.find_all([&](char c) {
```

Comment:
(Optional!) I don't think this `find_all()` actually simplifies the code a whole lot, I think you could do this as a simple for loop counting the number of fields so far as you go.  This would avoid allocating a vector here :^)

Reply:
It seems better since there is a need to check the total number of fields to ensure `nfields` is not too big (and thus needs a vector).

Reply:
Seems to actually complexify things since we don't know in the loop whether `nfields` is too big or that it's just not reached yet.

Reply:
I may be wrong, but I thought the logic of this could was roughly: 

```c++
bool in_field = false;
int field_number = 0;
int last_field_start = 0;

for (size_t i = 0; i < line.length(); i++) {
    char c = line[i];
    if (c == ' ' || c == '\t' || c == '\v' || c == '\f' || c == '\r') {
      in_field = false;
    } else if (!in_field) {
      in_field = true;
      last_field_start = i;
      field_number++;
      if (field_number > field_skip_count)
        break;
    }
}

line = line.substring_view(last_field_start);
```
So once you've seen more than `nfields` that's were you take the substring from that point, and if `nfields` is greater than the number of fields I think this will take the substring from the last field, which I think is what your code does.
I could be totally wrong here though, I've not tested this code at all. 

Anyway this is just a suggestion feel free to ignore me here :^)

Reply:
That's right, but is it really worth adding two more variables, one more condition and one more for loop just to avoid using .find_all?

Reply:
Well it's avoiding a heap allocating a vector for each line which depending on the data could be quite big I suppose

Reply:
Worth it